crates/common/validator/src/attestation.rs

```diff
@@ -14,3 +28,45 @@ pub fn compute_subnet_for_attestation(
     let committee_since_epoch_start = committees_per_slot * slots_since_epoch_start;
     (committee_since_epoch_start + committee_index) % ATTESTATION_SUBNET_COUNT
 }
+
+pub fn compute_on_chain_aggregate(
+    network_aggregates: Vec<Attestation>,
+) -> anyhow::Result<Attestation> {
+    let mut aggregates = network_aggregates.clone();
+
+    aggregates.sort_by(|a, b| {
+        let a_index = get_committee_indices(&a.committee_bits)[0];
+        let b_index = get_committee_indices(&b.committee_bits)[0];
+        a_index.cmp(&b_index)
+    });
+
+    let aggregation_bits_size = (MAX_VALIDATORS_PER_COMMITTEE * MAX_COMMITTEES_PER_SLOT) as usize;
+    let mut aggregation_bits = BitList::<U131072>::with_capacity(aggregation_bits_size)
+        .map_err(|_| anyhow::anyhow!("Failed to create BitList for aggregation_bits"))?;
+
+    for a in &aggregates {
+        for b in a.aggregation_bits.iter() {
+            let bit_index: usize = aggregation_bits.len();
+            aggregation_bits
+                .set(bit_index, b)
+                .map_err(|err| anyhow!("Failed to set bit {bit_index}: {err:?}"))?;
+        }
+    }
+    let signature_list: Vec<&BLSSignature> = aggregates.iter().map(|a| &a.signature).collect();
+    let committee_indices: HashSet<u64> = aggregates
+        .iter()
+        .map(|a: &Attestation| get_committee_indices(&a.committee_bits)[0])
```

Comment:
ideally we should be using `get()` instead of raw parsing `[]` as `[]` will panic on invalid inputs